eth/api.go

```diff
@@ -346,6 +347,40 @@ func (api *PrivateDebugAPI) GetBadBlocks(ctx context.Context) ([]*BadBlockArgs,
 	return results, nil
 }
 
+type AccountRangeResult struct {
+	result []common.Address `json:"result"`
+}
+
+func accountRange(st state.Trie, start *common.Address, maxResult int) (AccountRangeResult, error) {
+	it := trie.NewIterator(st.NodeIterator(start[:]))
+	result := []common.Address{}
+	for i := 0; i < maxResult && it.Next(); i++ {
+		key := st.GetKey(it.Key)
+		if key == nil {
+			return AccountRangeResult{}, fmt.Errorf("no preimage found for hash %x", it.Key)
+		}
+		result = append(result, common.BytesToAddress(key))
+	}
+	return AccountRangeResult{result}, nil
+}
+
+//block hash or number, tx index, start address hash, max results
+func (api *PrivateDebugAPI) AccountRange(ctx context.Context, blockNr uint64, txIndex int, startAddr *common.Address, maxResults int) (AccountRangeResult, error) {
+	zeros := make([]byte, common.HashLength)
+
+	if bytes.Equal(startAddr[:], zeros) {
+		startAddr = nil
+	}
+
+	blockHash := api.eth.blockchain.GetBlockByNumber(blockNr).Hash()
+	_, _, statedb, err := api.computeTxEnv(blockHash, txIndex, 0)
```

Comment:
I'm getting an error from this line:

```
{"jsonrpc":"2.0","id":1,"error":{"code":-32000,"message":"parent 0000000000000000000000000000000000000000000000000000000000000000 not found"}}
```

Not sure what I'm doing wrong here.

Reply:
I had a very quick look, and it seems that the message comes from here:
```go
        block := api.eth.blockchain.GetBlockByHash(blockHash)
        if block == nil {
                return nil, vm.Context{}, nil, fmt.Errorf("block %x not found", blockHash)
        }
        parent := api.eth.blockchain.GetBlock(block.ParentHash(), block.NumberU64()-1)
        if parent == nil {
                return nil, vm.Context{}, nil, fmt.Errorf("parent %x not found", block.ParentHash())
        }
```
it seems normal that block 0's parent will not be found. The `GetBlock` function uses both the hash and the block number to find the parent, and the block number will be invalid.